src/rulesets/Base/arraymath.jl

```diff
@@ -73,6 +73,54 @@ function rrule(
 end
 
 
+#####
+##### `muladd`
+#####
+
+function rrule(
+        ::typeof(muladd),
+        A::AbstractMatrix{<:CommutativeMulNumber},
+        B::AbstractVecOrMat{<:CommutativeMulNumber},
+        z::Union{CommutativeMulNumber, AbstractVecOrMat{<:CommutativeMulNumber}},
+    )
+    # The useful case, mul! fused with +
+    function muladd_pullback_1(Ȳ)
+        matmul = (
+            InplaceableThunk(
+                @thunk(Ȳ * B'),
+                dA -> mul!(dA, Ȳ, B', true, true)
+            ),
+            InplaceableThunk(
+                @thunk(A' * Ȳ),
+                dB -> mul!(dB, A', Ȳ, true, true)
+            )
+        )
+        addon = if z isa Bool
```

Comment:
```suggestion
        addon = if z isa Bool  # Bools are not considered as embedded in the Reals.
```

Reply:
Yea I was about to tag this. Does ChainRules agree with this policy? 

And is it able to test it? When I tried I got failures.

Reply:
Yes, we agree with this policy.
THough rather than `Zero` it should be `DoesNotExist`.
The subtype of `AbstractZero` which we use to indicate not only is it zero, but that that type fundermentally can't have a tangent space because the notion of perturbing it isn't defined.

And then we can test it by passing `(z, nothing)` to `rrule_test`, rather than `(z, dz)`

Reply:
Done